### PR TITLE
Ddcnl 6512

### PR DESCRIPTION
**Pull request by Nj147**

### CODE REVIEW

File: app/controllers/auth/AuthJourney.scala [hunk 1]
```diff
@@ -31,16 +31,17 @@ class AuthJourneyImpl @Inject() (
...
-    authAction andThen selfAssessmentStatusAction andThen getPersonDetailsAction
+    authAction andThen pertaxAuthAction andThen selfAssessmentStatusAction andThen getPersonDetailsAction
 
   override val authWithSelfAssessment: ActionBuilder[UserRequest, AnyContent] =
-    authAction andThen selfAssessmentStatusAction
+    authAction andThen pertaxAuthAction andThen selfAssessmentStatusAction
 
   override val minimumAuthWithSelfAssessment: ActionBuilder[UserRequest, AnyContent] =
-    minimumAuthAction andThen selfAssessmentStatusAction
+    minimumAuthAction andThen pertaxAuthAction andThen selfAssessmentStatusAction
```
Comment (pascal22p) [inline app/controllers/auth/AuthJourney.scala:45]: Double check but I think you should not use pertaxAuthAction here

**Review by pascal22p — CHANGES_REQUESTED** [review]

File: it/HomeControllerISpec.scala [hunk 2]
```diff
@@ -35,7 +35,8 @@ class HomeControllerISpec extends IntegrationSpec {
       "feature.business-hours.Saturday.start-time"     -> "0:00",
       "feature.business-hours.Saturday.end-time"       -> "23:59",
       "feature.business-hours.Sunday.start-time"       -> "0:00",
-      "feature.business-hours.sunday.end-time"         -> "23:59"
+      "feature.business-hours.sunday.end-time"         -> "23:59",
```
[the remark this answers is not in the document]
Comment (Nj147) [reply it/HomeControllerISpec.scala]: Yep I'd removed the classes but not he conf related stuff, done that now

File: app/controllers/auth/AuthJourney.scala (hunk 1 above)
Comment (Nj147) [reply app/controllers/auth/AuthJourney.scala]: Doesn't look like it should be, removed

File: app/connectors/PertaxConnector.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  appConfig: ConfigDecorator,
+  headerCarrierForPartialsConverter: HeaderCarrierForPartialsConverter
+)(implicit ec: ExecutionContext)
+    extends Logging {
+  private val pertaxUrl = appConfig.pertaxUrl
+
+  def pertaxAuthorise(
+    nino: String
+  )(implicit hc: HeaderCarrier): EitherT[Future, UpstreamErrorResponse, PertaxResponse] =
+    EitherT(
```
[the remark this answers is not in the document]
Comment (Nj147) [reply app/connectors/PertaxConnector.scala]: Done, had to add in some metrics for this connector too as that was a parameter for .read

File: app/controllers/auth/AuthAction.scala [hunk 4]
```diff
@@ -132,14 +131,7 @@ class AuthActionImpl @Inject() (
...
           } yield updatedResult
-
-          (configDecorator.singleAccountEnrolmentFeature, businessHours.isTrue(LocalDateTime.now())) match {
-            case (true, true) =>
-              if (enrolmentsHelper.singleAccountEnrolmentPresent(enrolments)) updatedResult
-              else Future.successful(Redirect(configDecorator.taxEnrolmentDeniedRedirect(request.uri)))
-            case _            => updatedResult
-          }
-
+          updatedResult
```
Comment (Christian-QA) [inline app/controllers/auth/AuthAction.scala:134]: You can return just the for/yield rather than making it a lazy val and returning it after

**Review by Christian-QA — APPROVED** [review]

**Review by pascal22p — APPROVED** [review]

File: app/controllers/auth/AuthAction.scala (hunk 4 above)
Comment (Nj147) [reply app/controllers/auth/AuthAction.scala]: Done

**Review by Christian-QA — APPROVED** [review]

**Review by Christian-QA — APPROVED** [review]

**Pull request closed without merging**